src/Database/Esqueleto/Experimental.hs

```diff
@@ -1205,52 +1205,47 @@ withRecursive baseCase unionKind recursiveCase = do
   Q $ W.tell mempty{sdCteClause = [clause]}
   pure refFrom
 
-type family ToAliasT a where
-  ToAliasT (SqlExpr (Value a)) = SqlExpr (Value a)
-  ToAliasT (SqlExpr (Entity a)) = SqlExpr (Entity a)
-  ToAliasT (SqlExpr (Maybe (Entity a))) = SqlExpr (Maybe (Entity a))
-  ToAliasT (a, b) = (ToAliasT a, ToAliasT b)
-  ToAliasT (a, b, c) = (ToAliasT a, ToAliasT b, ToAliasT c)
-  ToAliasT (a, b, c, d) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d)
-  ToAliasT (a, b, c, d, e) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d, ToAliasT e)
-  ToAliasT (a, b, c, d, e, f) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d, ToAliasT e, ToAliasT f)
-  ToAliasT (a, b, c, d, e, f, g) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d, ToAliasT e, ToAliasT f, ToAliasT g)
-  ToAliasT (a, b, c, d, e, f, g, h) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d, ToAliasT e, ToAliasT f, ToAliasT g, ToAliasT h)
-
 -- Tedious tuple magic
 class ToAlias a where
+  type ToAliasT a :: *
```

Comment:
This is great - I was wondering how feasible this would be. Thanks for trying it out!